homeassistant/components/switch/unifi.py

```diff
@@ -0,0 +1,222 @@
+
+    @callback
+    def _data_to_save():
+        """Update storage with devices that has got its' POE off."""
+        return switches_off
+
+    for switch in switches.values():
+        if switch.port.poe_mode == 'off' and \
+           switch.client.mac not in switches_off:
+            switches_off[switch.client.mac] = {
+                'data': switch.client.raw,
+                'poe_mode': switch.poe_mode
+            }
+            store.async_delay_save(_data_to_save, SAVE_DELAY)
+
+        elif (switch.port.poe_mode != 'off' and
+              switch.client.mac in switches_off):
+            del switches_off[switch.client.mac]
+            store.async_delay_save(_data_to_save, SAVE_DELAY)
+
+    if new_switches:
+        async_add_entities(new_switches)
+
+
+class UniFiSwitch(SwitchDevice):
+    """Representation of a client that uses POE."""
+
+    def __init__(self, client, controller,
+                 request_controller_update, poe_mode):
+        """Set up switch."""
+        self.client = client
+        self.controller = controller
+        self.poe_mode = poe_mode
+        self.async_request_controller_update = request_controller_update
+
+    async def async_update(self):
+        """Synchronize state with controller."""
+        if self.poe_mode is None and self.port.poe_mode != 'off':
+            self.poe_mode = self.port.poe_mode
+        await self.async_request_controller_update(self.client.mac)
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self.client.hostname
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this switch."""
+        return 'poe-{}'.format(self.client.mac)
+
+    @property
+    def is_on(self):
+        """Return true if POE is active."""
+        return self.port.poe_mode != 'off'
+
+    @property
+    def available(self):
+        """Return if switch is available."""
+        return self.controller.available or \
+            self.client.sw_mac in self.controller.api.devices
+
+    async def async_turn_on(self, **kwargs):
+        """Enable POE for client."""
+        await self.device.async_set_port_poe_mode(
+            self.client.sw_port, self.poe_mode)
+
+    async def async_turn_off(self, **kwargs):
+        """Disable POE for client."""
+        await self.device.async_set_port_poe_mode(self.client.sw_port, 'off')
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {
+            'power': self.port.poe_power,
+            'received': self.client.wired_rx_bytes / 1000000,
+            'sent': self.client.wired_tx_bytes / 1000000
+        }
+        return attributes
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            'connections': {(CONNECTION_NETWORK_MAC, self.client.mac)},
```

Comment:
This is going to be cool. First time that we can get 2 different integrations representing the same device. We should hook it into the device tracker too!

Reply:
Device tracker is next step for this! But next PR, no more gigantic PRs ;)